### PR TITLE
Rpi3 support

### PR DESCRIPTION
**Pull request by diederikdehaas**
There are still items to be improved, but it is essentially working on my systems.
One particular point that needs to be improved is the serial console/bluetooth. It is working during install, but during normal boot after the install it is not. And we probably need to add a config option which determines what to do with the UART port as that is either serial or bluetooth, but not both.

But for the time being people can now use the installer on a Pi 3 (but do have to build it themselves for now)

So I'd say this partially fixes issue #375

### CODE REVIEW

File: scripts/etc/init.d/rcS [hunk 1]
```diff
@@ -173,11 +173,28 @@ tee < ${LOGFILE}.pipe $LOGFILE &
...
+# http://elinux.org/RPi_HardwareHistory#Board_Revision_History
+cpu_revision=$(grep Revision /proc/cpuinfo | cut -d " " -f 2 | sed 's/^1000//')
+case $cpu_revision in
+    0011|0014)
+        rpi_hardware_version="cm"
+        ;;
+    900092|900093)
+        rpi_hardware_version="0"
+        ;;
+    0002|0003|0004|0005|0006|0007|0008|0009|000d|000e|000f|0010|0012|0013|0015)
```
Comment (Mausy5043) [inline scripts/etc/init.d/rcS]: add `|900032` ?

**Comment by Mausy5043** [comment]
Looks good to me. Can't test it though.

File: scripts/etc/init.d/rcS (hunk 1 above)
Comment (diederikdehaas) [inline scripts/etc/init.d/rcS]: Done. See new commit.

**Comment by Mausy5043** [comment]
Just tested it.

**Comment by diederikdehaas** [comment]
Also updated the README.md to include RPi 3B.

**Comment by cherylquirion** [comment]
Done. May I just say that I love your work. ;) It is working great for me guys. Well done. This build is absolutely smoking on the RPi3.

I will do another complete build of our system from scratch over the weekend just to confirm.

**Comment by cherylquirion** [comment]
Did a 2nd pass yesterday. All working well for me.

**Comment by cherylquirion** [comment]
Anyone having issues with installing RTC? We just got a new board and are trying to get rtc running to no avail.

**Comment by aykevl** [comment]
Works well for me, on my new RPi 3B. I built the installer and copied all files in `bootfs` to the root FAT32 partition on the SD card (manually partitioned).
The README needs some updates though, only in the beginning version 3B is mentioned, further down only 1 and 2.

File: scripts/etc/init.d/rcS [hunk 2]
```diff
@@ -1076,14 +1094,21 @@ do
...
-                echo "device_tree="
+                echo "# to disable DeviceTree, uncomment the next line "
+                echo "#device_tree="
+            fi
+            if [ "${hv}" = "3" ] ; then
+                # we should probably add a config parameter for uart purpose
+                # as the statements below disables bluetooth which various/most
+                # people will prefer over the serial console
+                echo "# enable the serial console for the installed system"
+                echo "#device_tree="
```
Comment (bbinet) [inline scripts/etc/init.d/rcS]: Do you mean "enable_uart=1" here?

Comment (diederikdehaas) [inline scripts/etc/init.d/rcS]: Indeed. Thanks for spotting this :+1:

**Comment by aristosv** [comment]
Following the build guide I created a build of the latest raspbian-ua-netinst. I run these commands:

git clone https://github.com/debian-pi/raspbian-ua-netinst.git
./build.sh

Everything run without errors, and created the .zip file.

I put it on a microsd and into the rpi3. I only got the rainbow screen.
I put the same microsd on an rpi2 and it worked just fine.

What I'm I doing wrong here?

**Comment by Mausy5043** [comment]
Try this:
```
git clone https://github.com/debian-pi/raspbian-ua-netinst.git
git checkout v1.1.x
./build.sh
```

**Comment by aristosv** [comment]
root@debian:~/project# git checkout v1.1.x
fatal: Not a git repository (or any of the parent directories): .git

**Comment by aristosv** [comment]
oops, I had to go in the folder.

**Comment by m1dnight** [comment]
I have tried building the image from the branch `v1.1.x` and I get a kernel panic. I would give more information but it scrolls by so fast I can't show much.

Anyone else experiencing this?

**Comment by CWempe** [comment]
Yes, it is a known issue.
The workaround is to edit the config.txt.
https://github.com/debian-pi/raspbian-ua-netinst/issues/472

But I am not sure what the solution will be.
Will down-/upgrading the firmware or kernel solve this?

**Comment by Mausy5043** [comment]
Please retry with the current v1.1.x branch (w/ #475 merged).

**Comment by CWempe** [comment]
I just tried a version I compiled yesterday.
`raspbian-ua-netinst-20170718-gite45eac1.zip`

Bu still get the kernel panic. :(

`git log` show this:

```
commit e45eac12a09f1f8478a62ed320eda4b4d4a1717c
Author: Diederik de Haas <github@cknow.org>
Date:   Sun Jul 16 21:12:08 2017 +0200

    Add HWRNG kernel module back and explicitly load the module. (#479)

    It was built-in with kernel 4.4, but with 4.9 it is once again built as
    a kernel module (like it was before 4.4).

commit 55e2963758778329d77609ffbd17902584536333
Merge: 93d2a91 8bf1b33
Author: Diederik de Haas <github@cknow.org>
Date:   Sat Jul 15 17:41:18 2017 +0200

    Merge pull request #475 from Mausy5043/kernel-4.9

    Change to Raspbian Kernel 4.9
```

So I think I am using the right branch/version.

**Comment by Mausy5043** [comment]
@CWempe and with UART disabled it works (again)?

**Comment by diederikdehaas** [comment]
I (normally) do my initial install with a serial console, so I can see what happens, and the `enable_uart=1` prevents the installation from even starting. Disabling that line made the installation start.
Problem is that I have no clue what's currently happening, but I suspect that the reboot will fail as that enables the uart again.

It is very likely that we need to switch to properly using the overlay mechanism and figure out how to do things 'the new way'.

**Comment by diederikdehaas** [comment]
I have these issues open in tabs as they deal with it, but haven't had time to properly go through them:

https://github.com/RPi-Distro/repo/issues/31
https://github.com/RPi-Distro/repo/issues/22
https://github.com/raspberrypi/firmware/issues/553

They may also be dated as I opened those tabs a while ago, but probably worth taking a look

**Comment by diederikdehaas** [comment]
> I suspect that the reboot will fail as that enables the uart again.

And indeed. Reboot failed. Disabling the `enable_uart=1` and it booted up.

So my suggestion is to disable that line for now, create a new beta release so the new kernel can be tested more broadly which may also fix some issues, and create a new issue to tackle the uart/serial console problem so we can get it working again.

**Comment by Mausy5043** [comment]
PR #480 should fix this.

**Pull request merged**